### ISSUE TITLE
Hive dialect name error

### ISSUE THREAD

**Issue by hsheth2** [fivetran/great_expectations#6146]
**Describe the bug**

The Hive dialect name is `b'hive'`, which is of type bytes and not str. This causes issues in the dialect comparison here: https://github.com/great-expectations/great_expectations/blob/1af478a2eb5694ad90d723a057be1c8636b2497b/great_expectations/execution_engine/sqlalchemy_dialect.py#L27-L30

This issue applies to GE versions 0.15.23+. 

This appears to be a regression introduced by https://github.com/great-expectations/great_expectations/pull/5980. Note that `b'foo' == 'foo'` is False but `hash(b'foo') == hash('foo')` is True, which is probably why this code worked before.

**To Reproduce**

We're running into this issue from the DataHub project, which depends on GE as a library. Hopefully the stack trace below has enough context, but I can try to provide more if needed.

Stack trace that I see:

```
Traceback (most recent call last):
  File "/home/mayuri/git/acryldata/datahub-fork/metadata-ingestion/src/datahub/ingestion/source/ge_data_profiler.py", line 906, in _generate_single_profile
    batch = self._get_ge_dataset(
  File "/home/mayuri/git/acryldata/datahub-fork/metadata-ingestion/src/datahub/ingestion/source/ge_data_profiler.py", line 970, in _get_ge_dataset
    batch = ge_context.data_context.get_batch(
  File "/home/mayuri/git/acryldata/datahub-fork/metadata-ingestion/venv/lib/python3.8/site-packages/great_expectations/data_context/data_context/base_data_context.py", line 1153, in get_batch
    return self._get_batch_v2(
  File "/home/mayuri/git/acryldata/datahub-fork/metadata-ingestion/venv/lib/python3.8/site-packages/great_expectations/data_context/data_context/base_data_context.py", line 859, in _get_batch_v2
    return validator.get_dataset()
  File "/home/mayuri/git/acryldata/datahub-fork/metadata-ingestion/venv/lib/python3.8/site-packages/great_expectations/validator/validator.py", line 2427, in get_dataset
    return self.expectation_engine(
  File "/home/mayuri/git/acryldata/datahub-fork/metadata-ingestion/venv/lib/python3.8/site-packages/great_expectations/dataset/sqlalchemy_dataset.py", line 568, in __init__
    if self.engine.dialect.name.lower() == GESqlDialect.BIGQUERY:
  File "/home/mayuri/git/acryldata/datahub-fork/metadata-ingestion/venv/lib/python3.8/site-packages/great_expectations/execution_engine/sqlalchemy_dialect.py", line 31, in __eq__
    return self.value.lower() == other.value.lower()
```

**Expected behavior**

No crashes.

**Environment (please complete the following information):**
 - Operating System: Linux
 - Great Expectations Version: 0.15.23+ (repro'd on .23, .25, and .26)

**Additional context**

DataHub introduced a temporary fix: https://github.com/datahub-project/datahub/pull/5980.

**Comment by AFineDayFor** [comment]
Howdy @hsheth2 :wave: thanks for reaching out and raising this with us :bow:

We're raising this within the team. Thank you so very much for helping us in identifying this 🦠🔬

**Comment by NathanFarmer** [comment]
The fix for this is merged into `develop` and will be included as part of todays release (0.15.27). Thanks @hsheth2!

**Comment by hsheth2** [comment]
Thanks @NathanFarmer for the quick fix!
